src/RpcClient/TransactionManager.cs

```diff
@@ -162,39 +161,80 @@ public TransactionManager AddWitness(UInt160 scriptHash, params object[] paramet
         /// <summary>
         /// Verify Witness count and add witnesses
         /// </summary>
-        public TransactionManager Sign()
+        public async Task<Transaction> SignAsync()
         {
+            // wait for all queued work to complete
+            await fluentOperationsTask;
```

Comment:
if we call again `MakeTransaction` the queue will be empty?

Reply:
If you call `MakeTransaction` again, it will queue two new items.

Based on this + the earlier comment about using the Tx before `ValidUntilBlock` and `SystemFee` are set, I'm going to redesign this API a bit.